### ISSUE TITLE
Identify and log information that might help peers debugging force-close issues

### ISSUE THREAD

**Issue by C-Otto** [lightningnetwork/lnd#6363]
### Background

My node just force-closed a channel due to a HTLC that timed out. I'd love to figure out what exactly caused my peer to not fail the HTLC. In order to do this, I'd like to help my peer dig through the logs and find the issue.

What kind of information would I need to send to a peer running lnd? c-lightning? other implementations? Where do I get this information from?

### Your environment

* version of `lnd`: 0.14.1-beta

**Comment by Roasbeef** [comment]
> My node just force-closed a channel due to a HTLC that timed out. I'd love to figure out what exactly caused my peer to not fail the HTLC. 

So if your node was the one that force closed, then the issue was actually _downstream_. Like if you had a channel with Bob, sent an HTLC, then either they or someone else downstream blew up, you'd eventually force close.

**Comment by C-Otto** [comment]
Yes. I'm pretty sure my peer (next hop) didn't blow up. This issue is about providing helpful information so that my peer can figure out what went wrong on their end.

**Comment by Roasbeef** [comment]
> My motivation is that there are too many force-closes that shouldn't happen,

Agreed, I think one way to start to shuffle around more debug information would be using the `warning` message to pass back a notification that you or your peer had to go on chain. There's some overlap here with the old "unwrap the onion" idea, but in this case we just want some basic timing information w.r.t what actually happened (if there's an identifiable reason other than a node going offline for a period of time). 

Re force closes, one thing that would be interesting to tease out is: the the force closes happening in only one area or the route, or are they actually cascading? You'd expect them to just happen in one area, with the others cancelling HTLCs back in most cases. However a cascade event would indicate some implementation level bug. 

On our end, we recently fixed an issue where some messages would be lost internally when we go on chain, leading to the incoming channel not immediately cancelling the HTLC: https://github.com/lightningnetwork/lnd/pull/6250.

**Comment by BhaagBoseDK** [comment]
we definitely need better logging when a channel is force closed 
`Local unilateral close of ChannelPoint`
does not tell anything of value or further investigation.

**Comment by C-Otto** [comment]
@BhaagBoseDK see #5432. In here (6363) I'd like to concentrate on information that can be used collaborate with a peer, i.e. this is more than just having better debug logs.

**Comment by fiatjaf** [comment]
Should channels really be force-closed in case of trimmed HTLC timeouts?

**Comment by Roasbeef** [comment]
Suggestion for the output here: https://github.com/lightningnetwork/lnd/issues/6615#issue-1260638827

**Comment by Roasbeef** [comment]
> Should channels really be force-closed in case of trimmed HTLC timeouts?

IMO no, see this issue: https://github.com/lightningnetwork/lnd/issues/1226 

An ideal set up IMO would be: always be 100% economically rational and _only_ go to chain if the fee revenue for going on-chain exceeds the fee revenue plus the HTLC amount. This would also mean mostly not caring about dust HTLCs, since usually you pay more in chain fees to force close then sweep than the amount plus the fees.

**Comment by litbot-9000** [comment]
Closing as duplicate of #10317 which consolidates requests for better force close information. Unique details from this issue have been summarized there.
